swagger/smart_transfer_protocol_ws.yaml

```diff
@@ -0,0 +1,421 @@
+asyncapi: 2.5.0
+info:
+  title: ODF Smart Transfer Protocol
+  version: '0.1.0'
+  description: Open Data Fabric (ODF) extended protocol for dataset exchange between remote parties.
+  license:
+    name: Apache 2.0
+    url: 'https://www.apache.org/licenses/LICENSE-2.0'
+
+channels:
+
+  dataset/pull/request:
+    subscribe:
+      operationId: odf.dataset.pull.request
+      summary: Initiates dataset pull from the ODF server
+      security:
+        - jwt: []
+      message:
+        messageId: odf.dataset.DatasetPullRequest
+        name: DatasetPullRequest
+        contentType: application/json
+        payload:
+          type: object
+          required:
+            - datasetId
+            - knownHead          
```

Comment:
Should this be optional as we won't have any `head` when pulling for the first time?

Reply:
Good catch!